client/my-sites/media/main.jsx

```diff
@@ -44,14 +52,91 @@ module.exports = React.createClass( {
 		page( redirect );
 	},
 
+	openDetailsModal( item ) {
+		this.setState( { openedDetails: item } );
+	},
+
+	closeDetailsModal() {
+		this.setState( { openedDetails: null, editedItem: null } );
+	},
+
+	editImage() {
+		this.setState( { openedDetails: null, editedItem: this.state.openedDetails } );
+	},
+
+	onImageEditorCancel: function( imageEditorProps ) {
```

Comment:
I debated on that for a bit in https://github.com/Automattic/wp-calypso/pull/8689#pullrequestreview-4159039, but I'd be open to making the caller more knowledgeable about the editor, and keeping the callbacks clean provided we provide a few more examples of usage in the readme. I agree with @artpi though that this should be tackled in another PR.